xunit.runner.wpf/Impl/RemoteTestUtil.Connection.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Pipes;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows.Threading;
+using xunit.runner.data;
+using xunit.runner.wpf.ViewModel;
+
+namespace xunit.runner.wpf.Impl
+{
+    internal sealed partial class RemoteTestUtil : ITestUtil
+    {
+        private sealed class Connection : IDisposable
+        {
+            private NamedPipeClientStream _stream;
+            private Process _process;
+            private ClientReader _reader;
+
+            internal NamedPipeClientStream Stream => _stream;
+
+            internal ClientReader Reader => _reader;
+
+            internal Connection(NamedPipeClientStream stream, Process process)
+            {
+                _stream = stream;
+                _process = process;
+                _reader = new ClientReader(stream);
+            }
+
+            internal void Close()
+            {
+                if (_process != null)
+                {
+                    Debug.Assert(_stream != null);
+
+                    try
+                    {
+                        _stream.WriteAsync(new byte[] { 0 }, 0, 1);
```

Comment:
The intent here is to allow predictable shutdown for both the UI and worker process.  The UI when completed with the connection writes a single byte to the worker process signaling completion.  Shutdown of pipe after that point is expected (see WaitForClientDone in Program.cs).  I used the async version because non-async can block and hang the UI. 

In some ways this hand shake is covering for my lack of understanding about certain aspects of named pipes.  I'm having an email convo with Pat right now to work out some of the details.  Once I get clarity on a few issues this can likely be cleaned up a bit.